crypto/mime.go

```diff
@@ -33,11 +34,15 @@ func (keyRing *KeyRing) DecryptMIMEMessage(
 		return
 	}
 
-	body, attachments, attachmentHeaders, err := parseMIME(string(decryptedMessage.GetBinary()), verifyKey)
+	body, attachments, attachmentHeaders, sigErr, err := parseMIME(string(decryptedMessage.GetBinary()), verifyKey)
 	if err != nil {
 		callbacks.OnError(err)
 		return
 	}
+	if sigErr != nil {
+		callbacks.OnError(err) // This is wrong and should use OnVerified instead
```

Comment:
Should we change this behavior to use OnVerified instead? Or we just consider this function to be legacy and try to deprecate it ?

Reply:
I think we should deprecate the callback based one and have only one interface

Reply:
Probably if we want to keep the callbacks we should fix `OnVerified`

Reply:
I don't think we need the callbacks, at least in the android client it was anyway blocking until all attachments were processed, I don't know about the iOS client

Reply:
IIRC it was iOS adding this originally, so better double check with them